src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -429,4 +436,30 @@ private void startJnlpProc() throws Exception {
         });
     }
 
+    private static class MyWorkspaceListener extends WorkspaceListener {
+        public boolean notified = false;
+
+        @Override
+        public void beforeUse(Run r, FilePath workspace, TaskListener listener) {
+            notified = true;
```

Comment:
Better to print something to `listener` and then use `JenkinsRule.assertLogContains`.